geospaas_harvesting/crawlers.py

```diff
@@ -231,12 +237,42 @@ def _get_links(cls, html):
         return parser.links
 
 
+class PODAACCrawler(WebDirectoryCrawler):
+    """
+    Crawler for harvesting the data of PODAAC
+    """
+    LOGGER = logging.getLogger(__name__ + '.PODAACCrawler')
+    FOLDERS_SUFFIXES = ('/contents.html')
+    FILES_SUFFIXES = ('.nc', '.nc.gz')
+    EXCLUDE = ['?']
+
+    def get_download_url(self, resource_url):
+        return resource_url
+
+
+class ThreddsCrawler(WebDirectoryCrawler):
+    """
+    Crawler for harvesting the data which are provided by Thredds from met.no
+    """
+    LOGGER = logging.getLogger(__name__ + '.ThreddsCrawler')
+    FOLDERS_SUFFIXES = ('/catalog.html',)
+    FILES_SUFFIXES = ('.nc',)
+    # We exclude "EASE-Grid map projections" and "southern hemispheres" from harvesting provess
+    EXCLUDE = ['/thredds/', 'http', '_sh_polstere', 'ease', ]
+
+    def get_download_url(self, resource_url):
+        links = self._get_links(self._http_get(resource_url))
+        for link in links:
+            if "dodsC" in link:
```

Comment:
It would be safer to match the link against a regular expression.
At least make sure it ends with ".html", because right now you cut the last four characters blindly.

Reply:
Done in the next commit